manifests/qpid.pp

```diff
@@ -2,25 +2,34 @@
 class katello::qpid (
   $client_cert,
   $client_key,
-  $katello_user          = $::katello::user,
-  $candlepin_event_queue = $::katello::candlepin_event_queue,
+  $katello_user             = $::katello::user,
+  $candlepin_event_queue    = $::katello::candlepin_event_queue,
+  $candlepin_qpid_exchange  = $::katello::candlepin_qpid_exchange,
+  $qpid_hostname            = $::katello::qpid_hostname,
 ){
   if $katello_user == undef {
     fail('katello_user not defined')
   } else {
     Group['qpidd'] ->
     User<|title == $katello_user|>{groups +> 'qpidd'}
   }
+  exec { 'create candlepin qpid exchange migrate':
+    command   => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://${qpid_hostname}:5671' add exchange topic ${candlepin_qpid_exchange} --durable",
+    unless    => "qpid-config --ssl-certificate ${client_cert} --ssl-key ${client_key} -b 'amqps://${qpid_hostname}:5671' exchanges ${candlepin_qpid_exchange}",
+    path      => '/usr/bin',
+    require   => Service['qpidd'],
+    logoutput => true,
+  } ~>
```

Comment:
@ekohl I doubt its needed, but I wanted to get this out of puppet-certs before changing to the new qpid exchange type and fixing other stuff :)